core/src/main/java/org/eclipse/hono/util/HonoDeviceNotificationConstants.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import org.apache.qpid.proton.message.Message;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Constants used for device notification events.
+ */
+public final class HonoDeviceNotificationConstants {
+    /**
+     * Prevent construction.
+     */
+    private HonoDeviceNotificationConstants() {
+    }
+
+    /**
+     * The content type that is defined for device notification events.
+     */
+    public static final String CONTENT_TYPE_DEVICE_NOTIFICATION = "application/vnd.eclipse-hono-notification+json";
+
+    /**
+     * Enum that defines all valid notifications by their content-type.
+     */
+    public enum NotificationContentType {
+        DEVICE_NOTIFICATION(CONTENT_TYPE_DEVICE_NOTIFICATION);
+
+        private final String contentType;
+
+        NotificationContentType(final String contentType) {
+            this.contentType = contentType;
+        }
+
+        /**
+         * Helper method to check if a subject is a valid content type.
+         *
+         * @param subject The subject to validate.
+         * @return boolean {@code true} if the subject denotes a valid action, {@code false} otherwise.
+         */
+        public static boolean isValid(final String subject) {
+            return Arrays.stream(NotificationContentType.values()).filter(type -> type.contentType.equals(subject)).findFirst().isPresent();
+        }
+    }
+
+    /**
+     * JSON field name for specifying the source of the event.
+     */
+    public static final String FIELD_SOURCE = "src";
```

Comment:
these constants seem to be only relevant to `NotificationDeviceCommandReadyObject` so we should probably move them there.